core-api/src/main/java/org/silverpeas/core/reminder/Reminder.java

```diff
@@ -0,0 +1,376 @@
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.silverpeas.core.reminder;
+
+import org.silverpeas.core.SilverpeasExceptionMessages;
+import org.silverpeas.core.SilverpeasRuntimeException;
+import org.silverpeas.core.admin.user.model.User;
+import org.silverpeas.core.contribution.ContributionManager;
+import org.silverpeas.core.contribution.model.Contribution;
+import org.silverpeas.core.contribution.model.ContributionIdentifier;
+import org.silverpeas.core.date.TimeUnit;
+import org.silverpeas.core.persistence.Transaction;
+import org.silverpeas.core.persistence.TransactionRuntimeException;
+import org.silverpeas.core.persistence.datasource.model.jpa.BasicJpaEntity;
+import org.silverpeas.core.scheduler.Scheduler;
+import org.silverpeas.core.scheduler.SchedulerProvider;
+import org.silverpeas.core.scheduler.trigger.JobTrigger;
+
+import javax.persistence.Column;
+import javax.persistence.DiscriminatorColumn;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import javax.persistence.Inheritance;
+import javax.persistence.InheritanceType;
+import javax.persistence.NamedQueries;
+import javax.persistence.NamedQuery;
+import javax.persistence.Table;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+/**
+ * A reminder. A reminder is a notification that is sent to a given user at a specific datetime.
+ * A reminder can be automatically rescheduled, meaning that it is rescheduled at another datetime
+ * at each of its triggering until that another datetime isn't defined. This capability depends on
+ * the concrete type of the used reminder and it is based upon the return of the
+ * {@link #isSchedulable()} method.
+ * @author mmoquillon
+ */
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@DiscriminatorColumn(name = "reminderType")
+@Table(name = "sb_reminder")
+@NamedQueries(
+    {@NamedQuery(name = "remindersByUserId", query = "select r from Reminder r where r.userId = :userId"),
+        @NamedQuery(name = "remindersByContributionId",
+            query = "select r from Reminder r where r.contributionId = :contributionId"),
+        @NamedQuery(name = "remindersByContributionIdAndUserId",
+            query = "select r from Reminder r where r.userId = :userId and r.contributionId = " +
+                ":contributionId")})
+public abstract class Reminder extends BasicJpaEntity<Reminder, ReminderIdentifier> {
+
+  @Embedded
+  private ContributionIdentifier contributionId;
+  @Column(name = "userId", nullable = false, length = 40)
+  private String userId;
+  @Column(name = "text")
+  private String text;
+  @Column(name = "triggered")
+  private boolean triggered;
+  @Column(name = "trigger_datetime", nullable = false)
+  private OffsetDateTime triggerDateTime;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make this value-based field transient so it is not included in the serialization of this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS3437)